### PR TITLE
Add `--compression` flag to `influx write`

### LINKED ISSUE #2240: InfluxDB 2.0.5: New flag for `influx write`
**PR:** influxdata/influxdb#20846

* `influx write` now accepts an optional `--compression` flag. The valid values are "none" and "gzip"
* If `--compression` isn't specified, it will be inferred based on file extension (files ending with `.gz` are assumed to be gzip)
* The new feature works along with existing `influx write` conveniences:
   * CSV inputs can be gzipped. They will be auto-detected based on the `.csv.gz` extension if the flags aren't specified
   * URL inputs can return gzipped data. If the flag isn't specified, GZIP will be inferred if (1) the URL ends with `.gz` or (2) the response from the server has a `Content-Encoding: gzip` header

### PR DESCRIPTION
**Pull request by pierwill**
Closes #2240.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/reference/cli/influx/write/_index.md [hunk 1]
```diff
@@ -63,30 +63,31 @@ In **extended annotated CSV**, measurements, fields, and values and their types
...
+| Flag |                     | Description                                                                                  | Input type  | {{< cli/mapped >}}    |
+|:-----|:--------------------|:---------------------------------------------------------------------------------------------|:-----------:|:----------------------|
+| `-c` | `--active-config`   | CLI configuration to use for command                                                         | string      |                       |
+| `-b` | `--bucket`          | Bucket name (mutually exclusive with `--bucket-id`)                                          | string      | `INFLUX_BUCKET_NAME`  |
+|      | `--bucket-id`       | Bucket ID (mutually exclusive with `--bucket`)                                               | string      | `INFLUX_BUCKET_ID`    |
+|      | `--configs-path`    | Path to `influx` CLI configurations (default `~/.influxdbv2/configs`)                        | string      | `INFLUX_CONFIGS_PATH` |
+|      | `--compression`     | Input compression, either `none` or `gzip`. Defaults to `none` unless input ends with `.gz`. | string      |                       |
```
Comment (pierwill) [inline content/influxdb/v2.0/reference/cli/influx/write/_index.md:72]: This is the new line. ⬆️

Comment (sanderson) [inline content/influxdb/v2.0/reference/cli/influx/write/_index.md:72]:
```suggestion
|      | `--compression`     | Input compression (`none` or `gzip`, default is `none` unless input file ends with `.gz`. | string      |                       |
```

File: content/influxdb/v2.0/reference/cli/influx/write/_index.md [hunk 2]
```diff
@@ -63,30 +63,31 @@ In **extended annotated CSV**, measurements, fields, and values and their types
...
+| `-p` | `--precision`       | Precision of the timestamps (default `ns`)                                                   | string      | `INFLUX_PRECISION`    |
+|      | `--rate-limit`      | Throttle write rate (examples: `5 MB / 5 min` or `1MB/s`).                                   | string      |                       |
+|      | `--skip-verify`     | Skip TLS certificate verification                                                            |             |                       |
+|      | `--skipHeader`      | Skip first *n* rows of input data                                                            | integer     |                       |
+|      | `--skipRowOnError`  | Output CSV errors to stderr, but continue processing                                         |             |                       |
+| `-t` | `--token`           | Authentication token                                                                         | string      | `INFLUX_TOKEN`        |
+| `-u` | `--url`             | URL to import data from                                                                      | stringArray |                       |
 
 ## Examples
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/cli/influx/write/_index.md:93]: We need to add examples for both line protocol and CSV. Add the following examples in their respective sections as well as links in the Examples TOC.

~~~md
##### Write line protocol from a compressed file
```sh
# The influx CLI assumes files with the .gz extension use gzip compression 
influx write \
  --bucket example-bucket \
  --file path/to/line-protocol.txt.gz

# Specify gzip compression for gzipped files without the .gz extension
influx write \
  --bucket example-bucket \
  --file path/to/line-protocol.txt.comp \
  --compression gzip
```

##### Write annotated CSV data from a compressed file
```sh
# The influx CLI assumes files with the .gz extension use gzip compression 
influx write \
  --bucket example-bucket \
  --file path/to/data.csv.gz

# Specify gzip compression for gzipped files without the .gz extension
influx write \
  --bucket example-bucket \
  --file path/to/data.csv.comp \
  --compression gzip
```
~~~

**Review by sanderson — CHANGES_REQUESTED** [review]

**Review by sanderson — APPROVED** [review]
🚀

**Pull request merged**